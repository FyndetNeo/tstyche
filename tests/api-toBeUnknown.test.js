import { expect, test } from "@jest/globals";
import * as tstyche from "tstyche";
import { getFixtureUrl } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixtureUrl = getFixtureUrl("api-toBeUnknown");

test("'toBeUnknown' implementation", () => {
  expect(tstyche.expect).toHaveProperty("type.toBeUnknown", expect.any(Function));
});

test("toBeUnknown", async () => {
  const { exitCode, stderr, stdout } = await spawnTyche(fixtureUrl);

  expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
  expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

  expect(exitCode).toBe(1);
});
