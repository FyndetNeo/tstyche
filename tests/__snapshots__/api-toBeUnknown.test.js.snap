// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeUnknown: stderr 1`] = `
"Error: Type 'unknown' is not identical to type 'string'.

   7 |   expect(returnsUnknown()).type.toBeUnknown();
   8 | 
>  9 |   expect(returnsString()).type.toBeUnknown();
     |                                ^
  10 | });
  11 | 
  12 | test("is NOT unknown?", () => {

       at ./__typetests__/toBeUnknown.tst.ts:9:32 ❭ is unknown?

Error: Type 'unknown' is identical to type 'unknown'.

  13 |   expect(returnsString()).type.not.toBeUnknown();
  14 | 
> 15 |   expect(returnsUnknown()).type.not.toBeUnknown();
     |                                     ^
  16 | });
  17 | 

       at ./__typetests__/toBeUnknown.tst.ts:15:37 ❭ is NOT unknown?

"
`;

exports[`toBeUnknown: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

fail ./__typetests__/toBeUnknown.tst.ts
  × is unknown?
  × is NOT unknown?

Targets:    1 failed, 1 total
Test files: 1 failed, 1 total
Tests:      2 failed, 2 total
Assertions: 2 failed, 2 passed, 4 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
